### PR TITLE
test(coderd): enable `dbrollup` service for insights tests

### PR DESCRIPTION
**Pull request by mafredri**

### CODE REVIEW

**Comment by mafredri** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/coder/coder/12673?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#12674** <a href="https://app.graphite.dev/github/pr/coder/coder/12674?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12673** <a href="https://app.graphite.dev/github/pr/coder/coder/12673?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#12672** <a href="https://app.graphite.dev/github/pr/coder/coder/12672?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12671** <a href="https://app.graphite.dev/github/pr/coder/coder/12671?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12670** <a href="https://app.graphite.dev/github/pr/coder/coder/12670?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12669** <a href="https://app.graphite.dev/github/pr/coder/coder/12669?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12668** <a href="https://app.graphite.dev/github/pr/coder/coder/12668?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12667** <a href="https://app.graphite.dev/github/pr/coder/coder/12667?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12666** <a href="https://app.graphite.dev/github/pr/coder/coder/12666?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12665** <a href="https://app.graphite.dev/github/pr/coder/coder/12665?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#12664** <a href="https://app.graphite.dev/github/pr/coder/coder/12664?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mafredri and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: coderd/database/queries/insights.sql [hunk 1]
```diff
@@ -357,7 +357,8 @@ WITH
 		SELECT
 			COALESCE(
 				MAX(start_time) - '1 hour'::interval,
-				NOW() - '6 months'::interval
+				-- TODO(mafredri): Fix this, required for tests to pass.
```
Comment (johnstcn) [inline coderd/database/queries/insights.sql:360]: Follow-up PR?

File: coderd/insights_test.go [hunk 2]
```diff
@@ -1204,6 +1206,9 @@ func TestTemplateInsights_Golden(t *testing.T) {
 			templates, users, testData := prepareFixtureAndTestData(t, tt.makeFixture, tt.makeTestData)
 			client := prepare(t, templates, users, testData)
 
+			// TODO(mafredri): Remove the need for this.
+			time.Sleep(3 * time.Second)
+
```
Comment (johnstcn) [inline coderd/insights_test.go:1211]: I wonder if you could (ab)-use the locking mechanism you introduced elsewhere?
As in, start the rollup, and then try to acquire the lock. Once the lock is released, you know rollup is done.

**Review by johnstcn — APPROVED** [review]

**Review by mtojek — APPROVED** [review]

File: coderd/insights_test.go (hunk 2 above)
Comment (mafredri) [reply coderd/insights_test.go]: I rewrote this using a channel for capturing events from the `dbrollup` service.

File: coderd/database/queries/insights.sql (hunk 1 above)
Comment (mafredri) [reply coderd/database/queries/insights.sql]: Fixed by changing the hard-coded nature of this query to use actual data from `workspace_agent_stats`.

**Comment by mafredri** [comment]
### Merge activity

* **Mar 22, 12:43 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/coder/coder/12673) rebased this pull request as part of a merge.

**Pull request merged**